samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -1004,6 +1021,7 @@ class SamzaContainer(
     info("Initializing stream tasks.")
 
     taskInstances.values.foreach(_.initTask)
+    taskInstances.values.foreach(_.startContext)
```

Comment:
Maybe move the startContext/stopContext inside initTask and shutdownTask instead of exposing as a new API?

Reply:
Makes sense.